pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -300,7 +310,9 @@ func validatePipelineWorkspacesDeclarations(wss []PipelineWorkspaceDeclaration)
 }
 
 // validatePipelineParameterUsage validates that parameters referenced in the Pipeline are declared by the Pipeline
-func (ps *PipelineSpec) validatePipelineParameterUsage() (errs *apis.FieldError) {
+func (ps *PipelineSpec) validatePipelineParameterUsage(ctx context.Context) (errs *apis.FieldError) {
+	errs = errs.Also(PipelineTaskList(ps.Tasks).validateUsageOfDeclaredPipelineTaskParameters(ctx, "tasks"))
+	errs = errs.Also(PipelineTaskList(ps.Finally).validateUsageOfDeclaredPipelineTaskParameters(ctx, "finally"))
```

Comment:
do we have unit test for this line ?

Reply:
yes. added it now. PTAL. I added a test for a finally task as well.